### DISCUSSION TITLE
flows do not support cookies

### DISCUSSION THREAD

**Discussion by zzxzz12345** [postmanlabs/postman-flows#22] [Defect]
flows do not support cookies

**Comment by saswatds** [comment]
@zzxzz12345 Can you update your Feedback to let us know what your use-case is and how cookies are not working for you?

**Reply by shamasis** [reply]
@saswatds my hunch is that for subsequent requests, the cookies are not being persisted. I tried sending the "Postman Echo / Cookies" folder and after setting a cookie, the next getting cookie returned none set.

<img width="1195" alt="image" src="https://user-images.githubusercontent.com/232373/136833559-98d3b831-73ab-4299-9ca2-126be8569a56.png">

**Reply by zzxzz12345** [reply]
1. use Postman Interceptor intercept the Cookies from Chrome
2. create a Request,this Request can access website in login state
![image](https://user-images.githubusercontent.com/8338183/136886118-b419c0ac-369e-426a-b63c-90841c378957.png)

3. create a Flow 
4. request in flow access the same website lost login state
![image](https://user-images.githubusercontent.com/8338183/136886054-380e1dcc-59bc-45c2-b46b-9ad5b7d11970.png)

**Reply by EricHerlitz** [reply]
Either the pre-request scripts or tests runs when you are using flows, at least not at the moment.

**Comment by barrycarr** [comment]
I've experienced the same thing, with regards to cookies. 

The first step in my flow logs in to our backend and returns a token cookie. Subsequent step fails with a 401.  If I run the same two calls outside of a flow, they both work correctly.

**Comment by saswatds** [comment]
Understood. Yeah, currently we have not added support for sharing cookies across multiple different requests. I am adding this as an enhancement #34

**Comment by dschealler** [comment]
Here's a use-case:

I'm currently doing some work for a client wanting to integrate from an Ops back-end package into MYOBA. Currently implementing Back-end Customers => MYOBA Customers, and Back-end Suppliers => MYOBA Suppliers (Vendors)

I've been using Postman as a rapid-iteration test-bed to get the message workflow proven before I try implementing any bespoke EDI code.

Basic pattern on the MYOBA side is: Log in, insert new customer, update new customer, delete new customer, log out. Same for suppliers. Have my tests and my environment UAT variables all set up and working as expected, works totally fine when I process the steps manually.

But in Flows it's failing. The log in step is successful, but the insert new customer step fails. If I attach a terminal to the failure state, it's failing with an error stating that I'm not logged in. Pretty sure that's the cookie problem above.

Not the end of the world because I can just run the steps manually. _However_, when I hand this project off to my support team? It would've been extremely valuable to give them some documentation along the lines of: _If you get an error with customers, to verify, go to the client workspace in Postman, select the appropriate environment, and execute workflow ABCD to smoke test end-to-end with the safe, default dummy data._

I can still give them documentation for how to run the steps manually. Trick is, not everyone in the support team is as competent with Postman as the delivery team members. Anything that reduces friction for the support team is a win... But in this scenario I need the cookies working in the Flows to get that outcome.

**Reply by saswatds** [reply]
@dschealler Thanks for this detailed feedback.  Cookies integration was blocked on internal change in the execution environment. Now that the internal change has been done I should be able to make cookies work across multiple requests within a few days. I will keep everyone posted here when cookie support lands in production.

**Reply by RoeyMichaeli** [reply]
Any news?

**Reply by saswatds** [reply]
It turned out to be much harder to get it working. I am working with the API client folks and will get this sorted out quick. Thanks for the patience. This is on the top of my priorities.

**Reply by dschealler** [reply]
@saswatds I have no doubt! I've developed in this area before and I won't be surprized at all if it's turning out to be a huge PITA.

For reference, I've already closed off that project without this feature, so the urgency is gone on my end. But it would be a very very nice-to-have feature on the next API project that requires cookies as part of its authentication model.

**Reply by saswatds** [reply]
Hey @dschealler, really sorry, I forgot to give an update here. I added the support to be able to persist cookies in this release [Release 22.03.1](https://github.com/postmanlabs/postman-flows/releases/tag/v22.03.1) It's not the perfect solution, but it should work for 90% of the use-cases. Do test it out and let me know if it works.

**Reply by dschealler** [reply]
Thanks @saswatds, I'll check it out next time I'm doing a discovery run on an API. :)
